pkg/packagejson/parse.go

```diff
@@ -0,0 +1,37 @@
+package npm
+
+import (
+	"encoding/json"
+	"io"
+
+	"github.com/aquasecurity/go-dep-parser/pkg/types"
+	"golang.org/x/xerrors"
+)
+
+type packageDotJSON struct {
+	Name    string      `json:"name"`
+	Version string      `json:"version"`
+	SHA     string      `json:"_shasum"`
+	License interface{} `json:"license"`
+}
+
+func Parse(r io.Reader) ([]types.Library, error) {
+	libs := []types.Library{}
+	var data packageDotJSON
+	err := json.NewDecoder(r).Decode(&data)
+	if err != nil {
+		return libs, xerrors.Errorf("decode error: %w", err)
+	}
+
+	// the license isn't always a string, so only take it if it is a string
+	license, _ := data.License.(string)
```

Comment:
I have a few questions.

License specification.
https://docs.npmjs.com/cli/v7/configuring-npm/package-json#license

There are five main ways to express a license.

1. One license
```
{
  "license" : "BSD-3-Clause"
}
```

2. SPDX license expression syntax (version 2.0)
```
{
  "license" : "(ISC OR GPL-3.0)"
}
```

3. External file reference 
```
{
  "license" : "SEE LICENSE IN <filename>"
}
```

4. Legacy format 
```
// Not valid metadata
{
  "license" : {
    "type" : "ISC",
    "url" : "https://opensource.org/licenses/ISC"
  }
}
// Not valid metadata
{
  "licenses" : [
    {
      "type": "MIT",
      "url": "https://www.opensource.org/licenses/mit-license.php"
    },
    {
      "type": "Apache-2.0",
      "url": "https://opensource.org/licenses/apache2.0.php"
    }
  ]
}
```

Is this pull request intended to address specifications 1 and 2?
As for 3, I think we could either skip it or parse.
As for 4, I'd like to discuss it, but is it possible to convert it to SPDX?

Reply:
For case 1, 2 & 3 currently complete string value will be picked. 
Legacy format is not supported, but will update code to support it.

Reply:
In the case of "SEE IN $filename", is it OK to just use the string as is, instead of going to reference it?